File: src/4-Problem02.py
if __name__ == "__main__":
    warna = []
    barang = []

    n_warna = int(input("Masukkan jumlah warna : "))
    print("Masukkan nama warna :")
    for i in range(n_warna) :
        warna.append(str(input()))

    n_barang = int(input("Masukkan jumlah barang : "))
    print("Masukkan nama barang :")
    for i in range(n_barang) :
        barang.append(str(input()))

    print("Ide barang selanjutnya ;")
    for bar in barang :
        for war in warna :
            print(bar, war)
    print("Terdapat {} ide yang mungkin".format(n_barang * n_warna))

File: src/4-Problem02.cpp
#include <iostream>

using namespace std;

int main(){
	/* Kamus */
	int banyak_warna, banyak_barang;
	string warna[10002], barang[10002];
	
	/* Algoritma */
	cout<<"Masukkan jumlah warna : ";
	cin>>banyak_warna;
	cout<<"Masukkan nama warna :"<<endl;
	for(int i=0;i<banyak_warna;++i){
		cin>>warna[i];
	}
	cout<<"Masukkan jumlah barang : ";
	cin>>banyak_barang;
	cout<<"Masukkan nama barang :"<<endl;
	for(int i=0;i<banyak_barang;++i){
		cin>>barang[i];
	}
	
	cout<<"Ide barang selanjutnya :"<<endl;
	for(int i=0;i<banyak_barang;++i){
		for(int j=0;j<banyak_warna;++j){
			cout<<barang[i]<<" "<<warna[j]<<endl;
		}
	}
	cout<<"Terdapat "<<banyak_barang*banyak_warna<<" ide yang mungkin"<<endl;
	return 0;
}